### ISSUE TITLE
__builtin_dump_struct doesn't handle float fields correctly

### ISSUE THREAD

**Issue by ndkrempel** [llvm/llvm-project#44488]
|  |  |
| --- | --- |
| Bugzilla Link | [45143](https://llvm.org/bz45143) |
| Version | 7.0 |
| OS | All |
| CC | @Quuxplusone,@DougGregor,@zygoloid |

## Extended Description 
The following program:

```C++
#include <cstdio>
struct Foo {
    float a;
};
int main() {
    Foo f = {1.2f};
    __builtin_dump_struct(&f, &std::printf);
}
```

prints out:

```
struct Foo {
float a : 0.000000
}
```

Rather than the correct value `1.2`.

Reproduced in clang 7.0.0, 8.0.0, 9.0.0, and trunk.

**Comment by ndkrempel** [comment]
This seems to have been fixed starting in clang version 15.0.0, where the correct `float` value is output.

**Comment by tbaederr** [comment]
Yep, this works now.
